### PR TITLE
fix(core): handle dependencies versions marked as exotic

### PR DESCRIPTION
**Pull request by prescottprue**
* fix(tests): add tests for getNumberOfDependencies
* fix(core): correctly calculate in date package percentage

## Notes
Fixes a case where an error was caused by dependencies marked as "exotic" - that happens if the dependency is being loaded from a repo instead of from the package registry

Here is the error happening in horchata with gs (which we have pointed to a private github repo instead of a package version)

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
         run: |
           echo "Uploading results to cloud storage"
           echo ""
-          bucket=reside-stage.appspot.com
+          bucket=some-project.appspot.com
```
Comment (hershmire) [inline README.md:116]: Why the change here?

**Review by hershmire — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (hershmire) [reply README.md:116]: Nm, that's the readme. Makes sense.

File: src/getDependencyStats.ts [hunk 2]
```diff
@@ -220,7 +229,11 @@ export default async function getDepedencyStats(): Promise<StatsOutput> {
   const majorPercentOutOfDate = ((majorsOutOfDate / numDeps) * 100).toFixed(2);
   const minorPercentOutOfDate = ((minorsOutOfDate / numDeps) * 100).toFixed(2);
   const patchPercentOutOfDate = ((patchesOutOfDate / numDeps) * 100).toFixed(2);
-  const upToDatePercent = ((majorsOutOfDate / numDeps) * 100).toFixed(2);
+  const upToDatePercent = (
+    ((numDeps - (majorsOutOfDate + minorsOutOfDate + patchesOutOfDate)) /
```
Comment (prescottprue) [inline src/getDependencyStats.ts:233]: @hershmire This was where the incorrectly calculated number was coming from - found it when adding more tests, and that case is now covered
